ambry-server/src/main/java/com/github/ambry/server/ParticipantsConsistencyChecker.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright 2020 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.server;
+
+import com.github.ambry.clustermap.ClusterParticipant;
+import com.github.ambry.commons.ServerMetrics;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * A thread that helps periodically check consistency of participants. Specifically, it checks if there is a mismatch
+ * in sealed and stopped replicas from each participant.
+ */
+public class ParticipantsConsistencyChecker implements Runnable {
+  private final List<ClusterParticipant> participants;
+  private final ServerMetrics metrics;
+  private static final Logger logger = LoggerFactory.getLogger(ParticipantsConsistencyChecker.class);
+
+  public ParticipantsConsistencyChecker(List<ClusterParticipant> participants, ServerMetrics metrics) {
+    this.participants = participants;
+    this.metrics = metrics;
+    metrics.registerParticipantsMismatchMetrics();
+  }
+
+  @Override
+  public void run() {
+    // when code reaches here, it means there are at least two participants.
+    ClusterParticipant clusterParticipant = participants.get(0);
+    Set<String> sealedReplicas1 = new HashSet<>(clusterParticipant.getSealedReplicas());
+    Set<String> stoppedReplicas1 = new HashSet<>(clusterParticipant.getStoppedReplicas());
+    for (int i = 1; i < participants.size(); ++i) {
+      Set<String> sealedReplicas2 = new HashSet<>(participants.get(i).getSealedReplicas());
+      if (!sealedReplicas1.equals(sealedReplicas2)) {
+        logger.warn("Mismatch in sealed replicas. Set {} is different from set {}", sealedReplicas1, sealedReplicas2);
+        metrics.sealedReplicasMismatchCount.inc();
```

Comment:
If this task is going to run every X seconds, do we want to increment the counter each time?  Once we have a mismatch, the metric count will increase proportionally to how often the schedule is run, which doesn't seem very meaningful.  Wondering if it should be a Guage instead.
Btw, what action can be taken if an alert on this metric is seen?

Reply:
Using counter seems inappropriate but is actually ok because Ingraph converts it to `per second rate` when it is read. (Reference: https://iwww.corp.linkedin.com/wiki/cf/display/SOP/inGraphs+FAQ#inGraphsFAQ). An example is `instanceConfigChangeTriggerCount`, if we look at the dashboard, it actually records number of events per second.

Reply:
If alert is fired, we can try to restart the server and attempt to correct the number. Sometimes the inconsistency comes from intermittent failure due to connection issue. 
If it doesn't work, we can use "partition override" feature to override certain partition to the correct state.

Reply:
I understand that counter is converted to rate.  My point was that the reported rate would be governed by how often the checker task kicks off, rather than how frequently an error happened or how many discrepancies there are, which might be confusing.  It's not a deal breaker, just wanted to point that out.